src/Command/Push/PushArtifactCommand.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+namespace Acquia\Cli\Command\Push;
+
+use Acquia\Cli\Command\Pull\PullCommandBase;
+use Acquia\Cli\Exception\AcquiaCliException;
+use Acquia\Cli\Output\Checklist;
+use Closure;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Finder\Finder;
+use Webmozart\PathUtil\Path;
+
+/**
+ * Class PushArtifactCommand.
+ */
+class PushArtifactCommand extends PullCommandBase {
+
+  protected static $defaultName = 'push:artifact';
+
+  /**
+   * @var string
+   */
+  protected $dir;
+
+  /**
+   * {inheritdoc}.
+   */
+  protected function configure(): void {
+    $this->setDescription('Build and push a code artifact to a Cloud Platform environment')
+      ->addOption('dir', NULL, InputArgument::OPTIONAL, 'The directory containing the Drupal project to be pushed')
+      ->addOption('no-sanitize', NULL, InputOption::VALUE_NONE, 'Do not sanitize the build artifact')
+      ->addOption('dry-run', NULL, InputOption::VALUE_NONE, 'Do not push changes to Acquia Cloud')
+      ->acceptEnvironmentId()
+    ->setHelp("This command builds a sanitized deploy artifact by running <options=bold>composer install</>, removing sensitive files, and committing vendor directories.\n\n"
+      . "The following vendor files and directories are committed to the build artifact even if they are ignored in the source repository: " . implode(', ', self::vendorFiles()) . "\n\n"
+      . "To run additional build or sanitization steps (e.g. <options=bold>npm install</>), add a <options=bold>post-install-cmd</> script to your <options=bold>composer.json</> file: https://getcomposer.org/doc/articles/scripts.md#command-events");
+  }
+
+  /**
+   * @param \Symfony\Component\Console\Input\InputInterface $input
+   * @param \Symfony\Component\Console\Output\OutputInterface $output
+   *
+   * @return int 0 if everything went fine, or an exit code
+   * @throws \Exception
+   */
+  protected function execute(InputInterface $input, OutputInterface $output): int {
+    // @todo change deploy strategy depending on whether this is a "source" repo (requiring building) or not
+    // @todo handle if Git user name/email is missing
+    $this->setDirAndRequireProjectCwd($input);
+    $is_dirty = $this->isLocalGitRepoDirty();
+    $commit_hash = $this->getLocalGitCommitHash();
+    if ($is_dirty) {
+      throw new AcquiaCliException('Pushing code was aborted because your local Git repository has uncommitted changes. Please either commit, reset, or stash your changes via git.');
+    }
+    $this->checklist = new Checklist($output);
+
+    // @todo handle environments with tags deployed
+    $this->logger->warning('You must select an environment with a Git branch deployed');
```

Comment:
Why are we showing a warning here? Seems like this is more informational, part of the context for the prompt.

Reply:
Indeed it's just informational, but I could not find any less severe log method that would display in non-verbose output. Am I missing something?

Reply:
Oh duh, I should be using `$output`, not `$this->logger`